### PR TITLE
LG-3942 Disabled button active state styles

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Fixes disabled state styling when a button is focused or active

🎟 _Jira ticket:_ https://jira.mongodb.org/browse/LG-3942

Screenshot 2024-01-26 at 2 04 58 PM

### For bug fixes, new features & breaking changes

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: packages/button/src/Button.story.tsx [hunk 1]
```diff
@@ -154,6 +151,25 @@ XSmallSize.parameters = {
...
+InteractiveStates.parameters = {
+  generate: {
+    args: {
+      rightGlyph: undefined,
+      leftGlyph: <Icon glyph={'Cloud'} />,
+      children: 'MongoDB',
+      variant: 'default',
+    },
+    combineArgs: {
+      // @ts-expect-error - data-focus is not listed as a prop
```
Comment (bruugey) [inline packages/button/src/Button.story.tsx:164]: nit: data-hover

File: packages/button/src/Button/Button.styles.ts [hunk 2]
```diff
@@ -13,6 +13,10 @@ import {
 
 import { ButtonProps, Size, Variant } from '../types';
 
+const focusSelector = `&:focus-within, &[data-focus="true"]`;
```
Comment (shaneeza) [inline packages/button/src/Button/Button.styles.ts:16]: Should this be focus-visible?

Comment (TheSonOfThomp) [reply packages/button/src/Button/Button.styles.ts:16]: Ah, yeah good catch

**Review by shaneeza — COMMENTED** [review]
@TheSonOfThomp I left a comment in chromatic

**Comment by TheSonOfThomp** [comment]
> @TheSonOfThomp I left a comment in chromatic

@shaneeza Good catch 👍 Fixed!

**Review by shaneeza — APPROVED** [review]

**Pull request merged**